pkg/activator/net/revision_backends.go

```diff
@@ -259,30 +259,46 @@ func (rw *revisionWatcher) checkDests(dests []string) {
 	}
 }
 
-func (rw *revisionWatcher) runWithTickCh(tickCh <-chan time.Time) {
+func (rw *revisionWatcher) run(probeFrequency time.Duration) {
 	var dests []string
+
 	for {
-		select {
-		case <-rw.doneCh:
-			return
-		case x, ok := <-rw.destsChan:
-			if !ok {
+		// We dont need to tick for probes if we have no dests or we are in clusterIP mode
+		if dests == nil || rw.clusterIPHealthy {
+			select {
+			case <-rw.doneCh:
+				return
+			case x, ok := <-rw.destsChan:
+				if !ok {
+					return
```

Comment:
This is all too complex :-) what you can do is a much simpler thing:
```
timer := time.NewTimer(pF)
defer close(timer.C)
var tickCh  chan time.Time
for {
select <-tickCh:
 // Exit select
case d, ok := <- rw.DestChan: 
  if !ok { return}
  dests := d
  // exit select
case  <-done:
  return
}
rw.checkDests(dests) // or checkDests can simply return bool that that we can use.
if len(dests) == 0 || rw.clusterIPHealthy 
  tickCh = nil
} else {
  tickCh = timer.C
}
```

Reply:
I'll add my own spin, building on what Victor suggests above, but mostly based on what you have:
```
	var dests []string

	for {
		var tickCh chan time.Time
		if dests != nil && !rw.clusterIPHealthy {
			tickCh = time.After(probeFrequency)
		}
		select {
		case <-rw.doneCh:
			return
		case x, ok := <-rw.destsChan:
			if !ok {
				return
			}
			dests = x
		case <-tickCh:
		}
		rw.checkDests(dests)
	}
```

Reply:
And last final approach can be to use timer.Stop() when we need to stop ticking and timer.Reset() when we need to start ticking again.

Reply:
Updated so theres a single select path. I wanted to use defer timer.Stop() as time.After is called out in godoc specifically for being unperformant, but otherwise its mostly the same as @mattmoor style.